lib/mcp_client/server_sse.rb

```diff
@@ -204,31 +194,34 @@ def rpc_request(method, params = {})
     # @return [void]
     def rpc_notify(method, params = {})
       ensure_initialized
-      url_base = @base_url.sub(%r{/sse/?$}, '')
-      uri = URI.parse("#{url_base}/messages?sessionId=#{@session_id}")
-      rpc_http = Net::HTTP.new(uri.host, uri.port)
-      if uri.scheme == 'https'
-        rpc_http.use_ssl = true
-        rpc_http.verify_mode = OpenSSL::SSL::VERIFY_PEER
+      # Determine RPC endpoint for notifications (prefer SSE-provided endpoint)
+      base = @base_url.sub(%r{/sse/?$}, '')
+      rpc_ep = @mutex.synchronize { @rpc_endpoint } || "/messages?sessionId=#{@session_id}"
```

Comment:
you have to read the endpoint and the query from the first connection., it get send a plain text.